component/src/main/java/org/wso2/extension/siddhi/gpl/execution/streamingml/classification/hoeffdingtree/HoeffdingClassifierUpdaterStreamProcessorExtension.java

```diff
@@ -0,0 +1,457 @@
+                                    + " manual configuration is %s but found %s",
+                            noOfHyperParameters, (noOfParameters - minNoOfParameters)));
+                }
+
+            }
+            evolutionModel = new PrequentialModelEvaluation();
+            evolutionModel.reset(noOfClasses);
+        } else {
+            throw new SiddhiAppValidationException(String.format("Invalid number of attributes for "
+                            + "streamingml:updateHoeffdingTree. This Stream Processor requires at least %s ,"
+                            + "parameters namely, model.name, number_of_classes and %s features but found %s "
+                            + "parameters and %s features",
+                    minNoOfParameters, minNoOfFeatures, (attributeExpressionLength - noOfFeatures), noOfFeatures));
+        }
+        //set attributes for OutputStream
+        List<Attribute> attributes = new ArrayList<>();
+        attributes.add(new Attribute("accuracy", Attribute.Type.DOUBLE));
+        return attributes;
+    }
+
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor
+            processor, StreamEventCloner streamEventCloner, ComplexEventPopulater complexEventPopulater) {
+        synchronized (this) {
+            while (streamEventChunk.hasNext()) {
+                ComplexEvent complexEvent = streamEventChunk.next();
+                String classValue = classLabelVariableExecutor.execute(complexEvent).toString();
+                cepEvent = new double[noOfFeatures];
+                for (int i = 0; i < noOfFeatures - 1; i++) {
+                    try {
+                        cepEvent[i] = ((Number) featureVariableExpressionExecutors.get(i)
+                                .execute(complexEvent)).doubleValue();
+                    } catch (ClassCastException e) {
+                        throw new SiddhiAppRuntimeException(String.format("Incompatible attribute feature type"
+                                + " at position %s. Not of any numeric type. Please refer the stream definition "
+                                + "for Model[%s]", (i + 1), modelName));
+                    }
+                }
+
+                AdaptiveHoeffdingTreeModel model = AdaptiveHoeffdingModelsHolder.getInstance()
+                        .getHoeffdingModel(modelName);
+
+                double accuracy;
+                if (model.getClasses().size() == noOfClasses) {
+                    accuracy = model.evaluationTrainOnEvent(evolutionModel, cepEvent, classValue);
+                } else {
+                    model.trainOnEvent(cepEvent, classValue);
+                    accuracy = 0;
+                }
+                complexEventPopulater.populateComplexEvent(complexEvent, new Object[]{accuracy});
+            }
+            nextProcessor.process(streamEventChunk);
+        }
+
```

Comment:
Remove unnecessary line spaces. I have seen some other places as well. Please fix all of them.

Reply:
Fixed in [#443c34d3d915f17058f93a17967c3e60de33cd1a](https://github.com/wso2-extensions/siddhi-gpl-execution-streamingml/pull/16/commits/443c34d3d915f17058f93a17967c3e60de33cd1a)